mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,289 @@
+    screen_resolution : array-like of shape ``(2,)``
+        The resolution (in pixels) of the screen that the eyetracking data
+        was collected with. For example, ``(1920, 1080)`` for a 1920x1080
+        resolution display.
+
+    Attributes
+    ----------
+    onset : float
+        The onset of the calibration in seconds. If the calibration was
+        performed before the recording started, the onset will be ``0`` seconds.
+    model : str
+        A string, which is the model of the calibration that was applied. For
+        example ``'H3'`` for a horizontal only 3-point calibration, or ``'HV3'`` for a
+        horizontal and vertical 3-point calibration.
+    eye : str
+        The eye that was calibrated. For example, ``'left'``, or ``'right'``.
+    avg_error : float
+        The average error in degrees between the calibration points and the actual gaze
+        position.
+    max_error : float
+        The maximum error in degrees that occurred between the calibration points and
+        the actual gaze position.
+    positions : ndarray of float, shape ``(n_calibration_points, 2)``
+        The x and y coordinates of the calibration points.
+    offsets : ndarray of float, shape ``(n_calibration_points,)``
+        The error in degrees between the calibration position and the actual
+        gaze position for each calibration point.
+    gaze : ndarray of float, shape ``(n_calibration_points, 2)``
+        The x and y coordinates of the actual gaze position for each calibration point.
+    screen_size : array-like
+        The width and height (in meters) of the screen that the eyetracking data was
+        collected  with. For example ``(.531, .298)`` for a monitor with a display area
+        of 531 x 298 mm.
+    screen_distance : float
+        The distance (in meters) from the participant's eyes to the screen.
+    screen_resolution : array-like
+        The resolution (in pixels) of the screen that the eyetracking data was
+        collected with. For example, ``(1920, 1080)`` for a 1920x1080 resolution
+        display.
+    """
+
+    def __init__(
+        self,
+        onset=None,
+        model=None,
+        eye=None,
+        avg_error=None,
+        max_error=None,
+        positions=None,
+        offsets=None,
+        gaze=None,
+        screen_size=None,
+        screen_distance=None,
+        screen_resolution=None,
+    ):
+        super().__init__(
+            onset=onset,
+            model=model,
+            eye=eye,
+            avg_error=avg_error,
+            max_error=max_error,
+            screen_size=screen_size,
+            screen_distance=screen_distance,
+            screen_resolution=screen_resolution,
+        )
+        self["positions"] = positions
+        self["offsets"] = offsets
+        self["gaze"] = gaze
+
+    def __repr__(self):
+        """Return a summary of the Calibration object."""
+        onset = self.get("onset", "N/A")
```

Comment:
Hmm yeah I see. Yeah,  it's easy enough to make the values for most of those keys `None` or `np.nan`, and most of them are just informational and so having values of `np.nan` or `None` is not going to break anything. 

Maybe making the `'positions'`, `'gaze'`, and `'offsets'` keys obligatory could be a little annoying for users who do have the `'model'`  `'avg_error'` and `'max_error'` information but don't have the data for individual points, in which case they would have to do something like `np.empty(n_calibration_points, 2)`.

But I'm not sure yet how often someone will even create one of these instances from scratch instead of using an available helper function like `read_eyelink_calibration`, so it's hard to say if it will be an issue for people?

Reply:
Let's tell them not to create it from scratch and just make sure we do the right thing internally. This is how Forward, SourceSpaces, etc. have worked for the most part for years and it has been okay. So +1 for keeping as simple as possible and adding more checks later if needed